### DISCUSSION TITLE
SMTP Password Causes Boot Loop

### DISCUSSION THREAD

**Discussion by ghost** [authelia/authelia#3642] [Q&A] [answered]
Earlier today, I enabled 2FA on the email account (I use my Microsoft Account) I use in Authelia for SMTP. Later in the day I tried to access a service behind Authelia and I got a 500 error. After some digging, I found that when I go to the Authelia page, I got a 502 Error. I checked the logs in docker and it was working fine, or so I thought. It kept repeating the startup message over and over. I checked the logs and SMTP kept popping up, leading me to believe it was related to the 2FA update. I generated an app password in my outlook email and replaced the password in the config with the app password i just made. Everything started working.

I am not sure Authelia is supposed to show an error for this but I feel like it would make sense IMO.

**Comment by nightah** [comment, marked as answer]
Authelia should be logging the error and the actual response it's getting back from the mail server, something like the following:

```
level=error msg="Failure running the notification provider startup check: 535 5.7.139 Authentication unsuccessful, the request did not meet the criteria to be authenticated successfully. Contact your administrator.
```

You should easily be able to test this by changing back to your password instead of the app password.

The only instance it wouldn't is if you [disabled the start-up check](https://www.authelia.com/configuration/notifications/introduction/#disable_startup_check) which in turn wouldn't result in a pre-check as part of Authelia's boot up, which would mean this is actually a configuration issue.

**Comment by ghost** [comment]
This is what I see 
`time="2022-06-29T21:19:40-04:00" level=info msg="Log severity set to debug"
time="2022-06-29T21:19:50-04:00" level=warning msg="Configuration: configuration key 'authentication_backend.disable_reset_password' is deprecated in 4.36.0 and has been replaced by 'authentication_backend.password_reset.disable': this has been automatically mapped for you but you will need to adjust your configuration to remove this message"
time="2022-06-29T21:19:50-04:00" level=info msg="Authelia v4.36.1 is starting"
time="2022-06-29T21:19:50-04:00" level=info msg="Log severity set to debug"
`

**Reply by james-d-elliott** [reply]
There's nothing there to indicate the cause of your issue. You probably want to submit more logs or wait for them to occur, there should be an error or a line indicating it's listening.

**Reply by ghost** [reply]
I found this in the log file (the previous was the console)
`0" level=warning msg="Notifier SMTP client encountered error during cleanup: use of closed network connection"
time="2022-06-29T21:00:31-04:00" level=error msg="Failure running the notification provider startup check: 535 5.7.139 Authentication unsuccessful, the user credentials were incorrect>
time="2022-06-29T21:00:31-04:00" level=fatal msg="The following providers had fatal failures during startup: notification" stack="github.com/authelia/authelia/v4/internal/commands/roo>
time="2022-06-29T21:00:31-04:00" level=info msg="Storage schema is being checked for updates"
time="2022-06-29T21:00:31-04:00" level=info msg="Storage schema is already up to date"
`
I still believe that the console log should have showed this

**Reply by nightah** [reply]
So that's basically the error that I was referencing, Authelia is indeed logging the error and notifying you of a problem.

Also just for the sake of clarity enabling 2FA and the behaviour elicited through Authelia by virtue of your change is completely expected. Microsoft does not allow simple authentication (read: username and password) and requires app passwords to be utilised when 2FA is enabled, which you obviously found out the hard way.

Not sure what else we can do, considering the logs are notifying you that there is a problem.

**Reply by ghost** [reply]
Alright, I figured an error like that should have showed up in the console. Sorry if I wasted anyone's time.

**Reply by nightah** [reply]
Sorry, I'm not clear on what you mean by console? How are you running Authelia, as bare-metal,  docker container or some other method?
Also just so we understand how you are configured can you please show the log section of your configuration file for Authelia?

That's the section that begins with:
```
log:
````

**Reply by ghost** [reply]
Sorry if I am using the wrong terminology. When I say console, I mean I just typed "docker logs authelia". 

Log Config:

`log:
  level: info
  file_path: /config/logs/authelia.log`

**Reply by james-d-elliott** [reply]
See our logging configuration. You can achieve this if you configure it appropriately: https://www.authelia.com/configuration/miscellaneous/logging/#keep_stdout

**Reply by ghost** [reply]
That's perfect! Exactly what I needed thank you.
